### PR TITLE
Include test for edge-ami image type

### PR DESCRIPTION
**Pull request by mcattamoredhat**
This pull request includes:

Now that osbuild-composer is able to generate an edge AMI that can be uploaded and deployed on AWS, it would be useful to include a dedicated test script for automation.

ostree-ami-image.sh uses osbuild-composer to build edge-ami image type, then upload it to a S3 bucket and deploy an EC2 instance using that image.

- [ ] adequate testing for the new functionality or fixed issue
- [ ] adequate documentation informing people about the change such as
  - [ ] submit a PR for the [guides](https://github.com/osbuild/guides) repository if this PR changed any behavior described there: https://www.osbuild.org/guides/

### CODE REVIEW

File: test/cases/ostree-ami-image.sh [hunk 1]
```diff
@@ -0,0 +1,1105 @@
...
+for LOOP_COUNTER in $(seq 0 30); do
+    RESULTS="$(wait_for_ssh_up $BIOS_GUEST_ADDRESS)"
+    if [[ $RESULTS == 1 ]]; then
+        echo "SSH is ready now! 🥳"
+        break
+    fi
+    sleep 10
+done
+
+# With new ostree-libs-2022.6-3, edge vm needs to reboot twice to make the /sysroot readonly
```
Comment (henrywang) [inline test/cases/ostree-ami-image.sh:723]: I think the ami image doesn't need reboot again, you can double confirm with devel.

File: test/cases/ostree-ami-image.sh [hunk 2]
```diff
@@ -0,0 +1,1105 @@
...
+check_result
+
+# Terminate running instance
+greenprint "Terminate running BIOS EC2 instance"
+aws ec2 terminate-instances --instance-ids "${INSTANCE_ID}"
+# Deregister edge AMI image
+greenprint "De-register BIOS AMI"
+aws ec2 deregister-image --image-id "${AMI_ID}"
+
+##################################################################
```
Comment (henrywang) [inline test/cases/ostree-ami-image.sh:776]: The ami image imported should support both bios and uefi. So we don't need two test scenarios.

File: test/cases/ostree-ami-image.sh [hunk 3]
```diff
@@ -0,0 +1,1105 @@
...
+if [[ $IMPORT_STATUS != completed ]]; then
+  echo "Something went wrong with the snapshot. 😢"
+  exit 1
+else
+  greenprint "Snapshot imported successfully."
+fi
+
+
+# Import  keypair
+greenprint "Share ssh public key with AWS"
```
Comment (henrywang) [inline test/cases/ostree-ami-image.sh:605]: All the following things, like keypair, vpc, internet gateway, subnet, you have two choices to configure them for test running in parallel support.

1. each test uses its own above things, their name should be random
2. each test uses the same above things, you need check them existing first. If they exist, just ignore configuration.

File: test/cases/ostree-ami-image.sh [hunk 4]
```diff
@@ -0,0 +1,1105 @@
...
+
+[[customizations.user]]
+name = "admin"
+description = "Administrator account"
+password = "\$6\$GRmb7S0p8vsYmXzH\$o0E020S.9JQGaHkszoog4ha4AQVs3sk8q0DvLjSMxoxHBKnB2FBXGQ/OkwZQfW/76ktHd0NX5nls2LPxPuUdl."
+home = "/home/admin/"
+groups = ["wheel"]
+EOF
+
+# No RT kernel in Fedora
```
Comment (henrywang) [inline test/cases/ostree-ami-image.sh:918]: Save above.

File: test/cases/ostree-ami-image.sh [hunk 5]
```diff
@@ -0,0 +1,1105 @@
...
+if [[ "$ID" != "fedora" ]]; then
+    tee -a "$BLUEPRINT_FILE" > /dev/null << EOF
+[customizations.kernel]
+name = "${KERNEL_RT_PKG}"
+EOF
+fi
+
+# Add directory and files customization, and services customization for testing
```
Comment (henrywang) [inline test/cases/ostree-ami-image.sh:926]: You already configure CUSTOM_DIRS_FILES to false, so this part of configuration is not use able.
Please confirm with devel for ami image custom dir support.

File: test/cases/ostree-ami-image.sh [hunk 6]
```diff
@@ -0,0 +1,1105 @@
...
+INSTALL_HASH=$(curl "${PROD_REPO_URL}/refs/heads/${OSTREE_REF}")
+
+# Add instance IP address into /etc/ansible/hosts
+sudo tee "${TEMPDIR}"/inventory > /dev/null << EOF
+[ostree_guest]
+${BIOS_GUEST_ADDRESS}
+
+[ostree_guest:vars]
+ansible_python_interpreter=/usr/bin/python3
+ansible_user=admin
```
Comment (henrywang) [inline test/cases/ostree-ami-image.sh:749]: Where do you test ignition user? You should test ignition user here.

File: test/cases/ostree-ami-image.sh [hunk 7]
```diff
@@ -0,0 +1,1105 @@
...
+    -e ostree_commit="${UPGRADE_HASH}" \
+    -e sysroot_ro="$SYSROOT_RO" \
+    -e test_custom_dirs_files="$CUSTOM_DIRS_FILES" \
+    /usr/share/tests/osbuild-composer/ansible/check_ostree.yaml || RESULTS=0
+check_result
+
+# Final success clean up
+clean_up
+
+greenprint "🧹 Clean up AWS EC2 resources"
```
Comment (henrywang) [inline test/cases/ostree-ami-image.sh:1074]: you have clean_up function already, can you move all clean up things there?

File: test/cases/ostree-ami-image.sh [hunk 8]
```diff
@@ -0,0 +1,1105 @@
...
+check_result
+
+# Add instance IP address into /etc/ansible/hosts
+sudo tee "${TEMPDIR}"/inventory > /dev/null << EOF
+[ostree_guest]
+${UEFI_GUEST_ADDRESS}
+
+[ostree_guest:vars]
+ansible_python_interpreter=/usr/bin/python3
+ansible_user=admin
```
Comment (henrywang) [inline test/cases/ostree-ami-image.sh:1052]: test ignition user here.

**Review by henrywang — CHANGES_REQUESTED** [review]

File: test/cases/ostree-ami-image.sh [hunk 9]
```diff
@@ -0,0 +1,1105 @@
...
+
+# TO DO: AWS configure with variables (github secrets)
+
+# Start a clean AWS cli installation 
+sudo rm -rf awscliv2.zip ./aws
+curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+unzip awscliv2.zip > /dev/null
+sudo ./aws/install --update
+aws --version
+
```
[the remark this answers is not in the document]
Comment (mcattamoredhat) [reply test/cases/ostree-ami-image.sh:482]: Ok, done.

**Review by schutzbot — COMMENTED** [review]
⚠️ This PR introduces changes in at least one manifest (when comparing PR HEAD 1b01accba508d2e5d0589f8b3c583dbb221cac38 with the main merge-base 1c148c162fcc7e7eeabb72c71b9ad41c3705c624).  Please review the changes.  The changes can be found in the [artifacts of the `Manifest-diff` job [0]](https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/4639894023/artifacts/browse) as `manifests.diff`.

[0] https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/4639894023/artifacts/browse

**Review by schutzbot — COMMENTED** [review]
⚠️ This PR introduces changes in at least one manifest (when comparing PR HEAD 1b01accba508d2e5d0589f8b3c583dbb221cac38 with the main merge-base 1c148c162fcc7e7eeabb72c71b9ad41c3705c624).  Please review the changes.  The changes can be found in the [artifacts of the `Manifest-diff` job [0]](https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/4639894999/artifacts/browse) as `manifests.diff`.

[0] https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/4639894999/artifacts/browse

**Review by schutzbot — COMMENTED** [review]
⚠️ This PR introduces changes in at least one manifest (when comparing PR HEAD cccc143399b5478b270399ce0faf703d67407a40 with the main merge-base 1c148c162fcc7e7eeabb72c71b9ad41c3705c624).  Please review the changes.  The changes can be found in the [artifacts of the `Manifest-diff` job [0]](https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/4653273883/artifacts/browse) as `manifests.diff`.

[0] https://gitlab.com/redhat/services/products/image-builder/ci/osbuild-composer/-/jobs/4653273883/artifacts/browse

**Review by henrywang — APPROVED** [review]
LGTM. Thanks!
